node/src/p2p/swarm.rs

```diff
@@ -1,73 +1,190 @@
+    use libp2p::{dns, noise, quic, swarm, tcp, websocket, yamux, PeerId, Transport};
+    use rustls_pki_types::{CertificateDer, PrivateKeyDer};
+    use tokio::fs;
+
     use super::*;
-    use libp2p::{dns, noise, tcp, yamux};
 
-    pub(crate) fn new_swarm<B>(keypair: Keypair, behaviour: B) -> Result<Swarm<B>, P2pError>
+    pub(crate) async fn new_swarm<B>(keypair: Keypair, behaviour: B) -> Result<Swarm<B>>
     where
         B: NetworkBehaviour,
     {
-        Ok(SwarmBuilder::with_existing_identity(keypair)
-            .with_tokio()
-            .with_tcp(
-                tcp::Config::default(),
-                noise::Config::new,
-                yamux::Config::default,
-            )?
-            .with_quic()
-            // We do not use system's DNS because libp2p loads DNS servers only when
-            // `Swarm` get constructed. This is not a problem for server machines, but
-            // it is for movable machines such as laptops and smart phones. Because of
-            // that, the following edge cases can happen:
-            //
-            // 1. Machine connects to WiFi A. Our node starts and uses DNS that
-            //    WiFi A announced. Then machine moves to WiFi B, the old DNS servers
-            //    become unreachable, but libp2p still uses the old ones.
-            // 2. Machine is not connected to the Internet. Our node starts and does not
-            //    find any DNS servers defined. Then machine connects to the Internet, but
-            //    libp2p still does not have any DNS nameservers defined.
-            //
-            // By having a pre-defined public servers, these edge cases solved.
-            .with_dns_config(
-                dns::ResolverConfig::cloudflare(),
+        let tls_key = match env::var("LUMINA_TLS_KEY_FILE") {
+            Ok(path) => Some(read_tls_key(path).await?),
+            Err(_) => None,
+        };
+
+        let tls_certs = match env::var("LUMINA_TLS_CERT_FILE") {
+            Ok(path) => Some(read_tls_certs(path).await?),
+            Err(_) => None,
+        };
+
+        // We do not use system's DNS because libp2p loads DNS servers only when
+        // `Swarm` get constructed. This is not a problem for server machines, but
+        // it is for movable machines such as laptops and smart phones. Because of
+        // that, the following edge cases can happen:
+        //
+        // 1. Machine connects to WiFi A. Our node starts and uses DNS that
+        //    WiFi A announced. Then machine moves to WiFi B, the old DNS servers
+        //    become unreachable, but libp2p still uses the old ones.
+        // 2. Machine is not connected to the Internet. Our node starts and does not
+        //    find any DNS servers defined. Then machine connects to the Internet, but
+        //    libp2p still does not have any DNS nameservers defined.
+        //
+        // By having a pre-defined public servers, these edge cases solved.
```

Comment:
```suggestion
        // We do not use system's DNS because libp2p caches system DNS servers when
        // `Swarm` get constructed, and doesn't update them later. This can be a problem,
        // if device roams between networks (and old DNS addresses may not be reachable from 
        // the new network). Similarly, if node is started when there's no Internet connection,
        // it won't use the DNS servers offered when Internet connectivity is restored.
        // Instead we per-define globally-accessible public DNS servers.
```
Reworded it to be slightly more concise, wdyt?

Reply:
looks good